src/components/AdminContainer.js

```diff
@@ -86,6 +86,11 @@ class AdminContainer extends Component {
     };
   }
 
+  componentDidMount() {
```

Comment:
does this mean that when you load the page through the normal selector page, this will run again too?

Reply:
yea

Reply:
will probably cause future problems if this solution of just adhoc retrying is implemented, 
i suggest you properly detect if the user hasn't selected an option

Reply:
updated, checked if the networkId is null

Reply:
just `if (networkId)`

checking against null is a bit of an antipattern in js